src/ldp/http/BasicRequestParser.ts

```diff
@@ -13,18 +14,18 @@ import type { TargetExtractor } from './TargetExtractor';
 export interface BasicRequestParserArgs {
   targetExtractor: TargetExtractor;
   preferenceParser: PreferenceParser;
-  metadataExtractor: MetadataExtractor;
+  metadataParser: MetadataParser;
   bodyParser: BodyParser;
 }
 
 /**
  * Creates an {@link Operation} from an incoming {@link HttpRequest} by aggregating the results
- * of a {@link TargetExtractor}, {@link PreferenceParser}, {@link MetadataExtractor}, and {@link BodyParser}.
+ * of a {@link TargetExtractor}, {@link PreferenceParser}, {@link MetadataParser}, and {@link BodyParser}.
  */
 export class BasicRequestParser extends RequestParser {
   private readonly targetExtractor!: TargetExtractor;
```

Comment:
`TargetParser` then too, I suppose?

Reply:
While this was more a result of removing the class `MetadataExtractor` and using its input classes directly (which were `MedataParser`s), that's not a bad idea.